scripts/ci-build-kubernetes.sh

```diff
@@ -0,0 +1,129 @@
+#!/bin/bash
+
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+###############################################################################
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+cd "${REPO_ROOT}" || exit 1
+
+# shellcheck source=../hack/ensure-go.sh
+source "${REPO_ROOT}/hack/ensure-azcli.sh"
+# shellcheck source=../hack/ensure-go.sh
+source "${REPO_ROOT}/hack/ensure-go.sh"
+# shellcheck source=../hack/parse-prow-creds.sh
+source "${REPO_ROOT}/hack/parse-prow-creds.sh"
+
+: "${AZURE_STORAGE_ACCOUNT:?Environment variable empty or not defined.}"
+: "${AZURE_STORAGE_KEY:?Environment variable empty or not defined.}"
+: "${REGISTRY:?Environment variable empty or not defined.}"
```

Comment:
REGISTRY is only set on line 71 in ci-entrypoint but we're calling this script earlier on line 45, you might have to move it up

Reply:
also don't we need to do stuff like https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/scripts/ci-conformance.sh#L60-L66 to sign into the capzci registry when using it from prow?

Reply:
Created hack/ensure-registry.sh (https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1264/files#diff-0aa6435fc476358f971605d450904f17c269e696052f9d339288ad5908716060R1-R33) inspired by https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1330